src/ddci.c

```diff
@@ -762,12 +797,22 @@ int ddci_add_psi(ddci_device_t *d, uint8_t *dst, int len) {
     int64_t ctime = getTick();
     int i, pos = 0;
     int psi_len;
+
+    // Add PAT
     if (ctime - d->last_pat > 500) {
         psi_len = ddci_create_pat(d, psi);
         pos += buffer_to_ts(dst + pos, len - pos, psi, psi_len, &d->pat_cc, 0);
         d->last_pat = ctime;
     }
 
+    // Add SDT
```

Comment:
Is best if this is after the PMT because the pmt->name can be populated based on the SDT.

Reply:
The virtual SDT doesn't contain any service names at the moment, only service IDs and some status flags.